app/controllers/checkin_controller.rb

```diff
@@ -1,12 +1,16 @@
 class CheckinController < ApplicationController
   before_action :require_login
 
-  def create
+  def new
     flash.clear
-    Time.zone = 'Pacific Time (US & Canada)'
     @user = Student.find_by_id(session[:current_user_id])
     @checkin = Checkin.new
-    @checkin.update(time: DateTime.now, student_id: @user.id)
+  end
+
+  def create
+    flash.clear
+    Time.zone = 'Pacific Time (US & Canada)'
+    @checkin.update(time: DateTime.now, student_id: @user.id, reason: params[:checkin][:reason])
```

Comment:
Because HTTP request is stateless, you may need to put `@checkin` and `@user` in session and retrieve them every time. 
![image](https://user-images.githubusercontent.com/56783069/159185119-d7c3cbc0-9dd9-43ac-b475-5e21355c94f9.png)